### PR TITLE
[examples][MLIRSparseTensor] add operation for sparse_tensor.expand/c…

### PR DESCRIPTION
**Pull request by Avimitin**
…ompress

### CODE REVIEW

File: examples/MLIRSparseTensor/sparse-tensor-expand.mlir [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+// @matmul will be rewrited into code like @matmul_expand.
+// It use `sparse_tensor.expand` to get the innermost level access pattern of the given tensor.
+// This operation is useful to implement calculation kernel on the output kernel.
+// The `values` array contains values in the innermost level of the given tensor.
+// The `filled` array contains boolean value that indicate whether a coordinate has been filled at the level.
+// The `added` array and `count` array are used to store new level-coordinate when the current operating coordinate has never been filled.
+// After operation, the rewrite op will use `sparse_tensor.compress` to compress the access pattern into the given tensor.
+func.func @matmul_expand(%A: tensor<4x2xf64, #CSR>, %B: tensor<2x4xf64, #CSR>) -> tensor<4x4xf64, #CSR> {
+    %c4 = arith.constant 4 : index
```
Comment (zhanghb97) [inline examples/MLIRSparseTensor/sparse-tensor-expand.mlir:23]: Please follow the same 2-indent style here.

File: examples/MLIRSparseTensor/sparse-tensor-expand.mlir [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  return %D: tensor<4x4xf64, #CSR>
+}
+
+// @matmul will be rewrited into code like @matmul_expand.
+// It use `sparse_tensor.expand` to get the innermost level access pattern of the given tensor.
+// This operation is useful to implement calculation kernel on the output kernel.
+// The `values` array contains values in the innermost level of the given tensor.
+// The `filled` array contains boolean value that indicate whether a coordinate has been filled at the level.
+// The `added` array and `count` array are used to store new level-coordinate when the current operating coordinate has never been filled.
+// After operation, the rewrite op will use `sparse_tensor.compress` to compress the access pattern into the given tensor.
```
Comment (zhanghb97) [inline examples/MLIRSparseTensor/sparse-tensor-expand.mlir:21]: The "After operation" here seems ambiguous, can we use the statement like "After expanding", or "After above operations"?

File: examples/MLIRSparseTensor/makefile [hunk 3]
```diff
@@ -44,3 +44,10 @@ sparse-tensor-pack-run:
 		--sparse-compiler="enable-runtime-library=false" | \
 	${MLIR_CPU_RUNNER} -e main -O0 --entry-point-result=void \
 		--shared-libs=${MLIR_RUNNER_UTILS},${MLIR_C_RUNNER_UTILS}
+
+# `sparse_tensor.expand/compress` are lowering operation
```
Comment (zhanghb97) [inline examples/MLIRSparseTensor/makefile:48]: The statement here seems ambiguous. Can we use a statement like "`sparse_tensor.expand/compress` operations are lowered from high-level linalg operation"?

**Review by zhanghb97 — APPROVED** [review]
LGTM.
Please fix some details, resolve the conflict, and push upstream.

**Comment by Avimitin** [comment]
All the required code changes are made. Ready for new review.

**Comment by zhanghb97** [comment]
Thanks!

**Pull request merged**